File: csharp/FiveCardStud.cs
using System;

// The FiveCardStud class serves as the main entry point for the poker hand analyzer program
class FiveCardStud
{
    // The Main method is the entry point of the program
    static void Main(string[] args)
    {
        // If no command-line arguments are provided, run the program in interactive mode
        if (args.Length == 0)
        {
            Console.WriteLine("*** POKER HAND ANALYZER ***\n");
            bool test = false;

            // Create an ordered deck of cards
            CreateDeck deck = new CreateDeck(test);

            // Shuffle the deck
            deck.ShuffleDeck();

            // Display the shuffled deck
            deck.RetrieveCard();

            // Deal six hands from the top of the deck
            deck.DealHands();

            Console.WriteLine();

            // Display the dealt hands
            deck.PrintHands();

            Console.WriteLine();

            // Show the remaining cards in the deck
            deck.RemainingDeck();

            // Print the final results (hand rankings)
            deck.PrintResult();
        }
        // If the number of command-line arguments is not 1, display usage information
        else if (args.Length != 1)
        {
            Console.Error.WriteLine("Usage: mono Program.exe <filename>");
        }
        // If exactly one command-line argument is provided, run the program in test mode
        else
        {
            Console.WriteLine(" *** P O K E R   H A N D  A N A L Y Z E R ***\n");
            Console.WriteLine("*** USING TEST DECK ***\n");

            // Get the filename from the command-line argument
            string filename = args[0];
            Console.WriteLine($"*** File: {filename}");

            // Create a new GameMode object to handle the test scenario
            GameMode mode = new GameMode();
            try
            {
                // Read the test data from the specified file
                mode.ReadFile(filename);
                // Create a new deck for the test scenario
                CreateDeck testDeck = new CreateDeck();
                // Retrieve the test deck data
                testDeck.RetrieveDeck(mode);
                // Process the test deck
                mode.CallDeckClass(testDeck);
                // Print the final results (hand rankings) for the test scenario
                testDeck.PrintResult();
            }
            catch (Exception e)
            {
                // If an error occurs during file processing, display error information
                Console.Error.WriteLine($"Error processing file: {e.Message}");
                Console.Error.WriteLine(e.StackTrace);
            }
        }
    }
}

File: java/FiveCardStud.java
/**
 * The FiveCardStud class serves as the main entry point for the Poker Hand Analyzer program.
 * It handles both normal gameplay and test mode operations.
 */
public class FiveCardStud {
    /**
     * The main method that runs the Poker Hand Analyzer.
     * It handles two modes of operation:
     * 1. Normal mode: Shuffles a deck, deals hands, and analyzes them.
     * 2. Test mode: Reads hands from a file and analyzes them.
     * 
     * @param args Command line arguments. If provided, should be a single filename for test mode.
     */
    public static void main(String[] args) {
        if (args.length == 0) {
            // Normal mode operation
            System.out.println("*** POKER HAND ANALYZER ***\n");
            boolean test = false;

            // Create an ordered deck of cards
            CreateDeck deck = new CreateDeck(test);

            // Shuffle the deck
            deck.shuffleDeck();

            // Display the shuffled deck
            deck.retrieveCard();

            // Deal six hands from the top of the deck
            deck.dealHands();

            System.out.println();

            // Print the dealt hands
            deck.printHands();

            System.out.println();

            // Show the remaining cards in the deck
            deck.remainingDeck();

            // Analyze and print the results
            deck.printResult();
        } else if (args.length != 1) {
            // Incorrect usage of command line arguments
            System.err.println("Usage: java FiveCardStud <filename>");
        } else {
            // Test mode operation
            System.out.println(" *** P O K E R   H A N D  A N A L Y Z E R ***\n");
            System.out.println("*** USING TEST DECK ***\n");

            String filename = args[0];
            System.out.println("*** File: " + filename);

            GameMode mode = new GameMode();
            try {
                // Read test hands from the specified file
                mode.readFile(filename);
                
                // Create a test deck and retrieve hands from the GameMode
                CreateDeck testDeck = new CreateDeck();
                testDeck.retrieveDeck(mode);
                
                // Process the test deck
                mode.callDeckClass(testDeck);
                
                // Analyze and print the results
                testDeck.printResult();
            } catch (Exception e) {
                // Handle any errors that occur during file processing
                System.err.println("Error processing file: " + e.getMessage());
                e.printStackTrace();
            }
        }
    }
}
